Src/Support/Google.Apis.Auth/OAuth2/DefaultCredentialProvider.cs

```diff
@@ -260,14 +261,64 @@ private static UserCredential CreateUserCredentialFromParameters(JsonCredentialP
             return new ServiceAccountCredential(initializer.FromPrivateKey(credentialParameters.PrivateKey));
         }
 
+        /// <summary>
+        /// Creates an external account credential from JSON data.
+        /// </summary>
+        private static IGoogleCredential CreateExternalCredentialFromParametes(JsonCredentialParameters parameters)
+        {
+            if (parameters.Type != JsonCredentialParameters.ExternalAccountCredentialType || parameters.CredentialSourceConfig is null)
+            {
+                throw new InvalidOperationException("JSON data does not represent a valid external account credential.");
+            }
+
+            // Build the external credential of the correct type.
+            // The order in which these checks are performed is relevant, see https://google.aip.dev/auth/4117.
+            if (!string.IsNullOrEmpty(parameters.CredentialSourceConfig.EnvironmentId))
+            {
+                throw new NotImplementedException("AWS external credentials not yet supported.");
+            }
+            else if (!string.IsNullOrEmpty(parameters.CredentialSourceConfig.File))
+            {
+                throw new NotImplementedException("File-sourced credentials not yet supported.");
+            }
+            else if (!string.IsNullOrEmpty(parameters.CredentialSourceConfig.Url))
+            {
+                var initializer = new UrlSourcedExternalAccountCredential.Initializer(
+                    parameters.TokenUrl, parameters.Audience, parameters.SubjectTokenType, parameters.CredentialSourceConfig.Url)
+                {
+                    QuotaProject = parameters.QuotaProject,
```

Comment:
We need to make sure the workforce_pool_user_project is being passed as well (for workforce flows).

[go/workforce-pools-client-support](http://go/workforce-pools-client-support)